cmd/update/project.go

```diff
@@ -4,89 +4,147 @@ import (
+         app: flyte
+    description: "Some description for the project"
 
+Update a project state by definition file. Note: The name shouldn't contain any whitespace characters.
 ::
 
- flytectl update project unknown-project --archiveProject -p known-project
+ flytectl update project --file project.yaml  --archive
+
+.. code-block:: yaml
+
+    id: "project-unique-id"
+    name: "Name"
+    labels:
+       values:
+         app: flyte
+    description: "Some description for the project"
 
 Usage
 `
 )
 
-var DefaultProjectConfig = &ProjectConfig{}
-
 func updateProjectsFunc(ctx context.Context, args []string, cmdCtx cmdCore.CommandContext) error {
-	id := config.GetConfig().Project
-	if id == "" {
-		fmt.Printf(clierrors.ErrProjectNotPassed)
-		return nil
+	projectSpec, err := util.GetProjectSpec(project.DefaultProjectConfig, config.GetConfig().Project)
+	if err != nil {
+		return err
+	}
+	if projectSpec.Id == "" {
+		return fmt.Errorf(clierrors.ErrProjectNotPassed)
+	}
+
+	projectDefinition := &admin.Project{
+		Id: projectSpec.Id,
+	}
+	if projectSpec.Description != "" {
+		projectDefinition.Description = projectSpec.Description
 	}
-	archiveProject := DefaultProjectConfig.ArchiveProject
-	activateProject := DefaultProjectConfig.ActivateProject
-	if activateProject == archiveProject {
-		return fmt.Errorf(clierrors.ErrInvalidStateUpdate)
+	if projectSpec.Name != "" {
+		projectDefinition.Name = projectSpec.Name
 	}
-	projectState := admin.Project_ACTIVE
-	if archiveProject {
-		projectState = admin.Project_ARCHIVED
+	if len(projectSpec.Labels.Values) > 0 {
+		projectDefinition.Labels = projectSpec.Labels
 	}
-	if DefaultProjectConfig.DryRun {
+
+	projectDefinition, err = getState(project.DefaultProjectConfig, projectDefinition)
+	if err != nil {
+		return err
+	}
+
+	if project.DefaultProjectConfig.DryRun {
 		logger.Infof(ctx, "skipping UpdateProject request (dryRun)")
 	} else {
-		_, err := cmdCtx.AdminClient().UpdateProject(ctx, &admin.Project{
-			Id:    id,
-			State: projectState,
-		})
+		_, err := cmdCtx.AdminClient().UpdateProject(ctx, projectDefinition)
 		if err != nil {
-			fmt.Printf(clierrors.ErrFailedProjectUpdate, id, projectState, err)
+			fmt.Printf(clierrors.ErrFailedProjectUpdate, projectSpec.Id, err)
 			return err
 		}
 	}
-	fmt.Printf("Project %v updated to %v state\n", id, projectState)
+	fmt.Printf("Project %v updated\n", projectSpec.Id)
 	return nil
 }
+
+func getState(flags *project.ConfigProject, spec *admin.Project) (*admin.Project, error) {
+	if flags.ActivateProject {
+		flags.Activate = flags.ActivateProject
+	}
+	if flags.ArchiveProject {
+		flags.Archive = flags.ArchiveProject
+	}
+
+	activate := flags.Activate
+	archive := flags.Archive
+
+	if activate || archive {
+		if activate == archive {
+			return spec, fmt.Errorf(clierrors.ErrInvalidStateUpdate)
+		}
+		spec.State = admin.Project_ACTIVE
+		if archive {
+			spec.State = admin.Project_ARCHIVED
+		}
+		return spec, nil
+	}
+	return spec, nil
```

Comment:
Instead of passing in the spec and also having to return the same object ,
We can have this (f *project.ConfigProject) func mapToAdminState() (Project_ProjectState, error) 
and have the spec assignment be done in the consumer call